### ISSUE TITLE
license-detector-and-feeder branch. Issue with mvn install for detector and feeder.

### ISSUE THREAD

**Issue by michelescarlato** [fasten-project/fasten#416]
@mir-am @MagielBruntink 
Hi Amir and Magiel,

Amir suggested [here](https://github.com/fasten-project/fasten/issues/408#issuecomment-1048554525), I am using a newer version of develop, but I am having issues with compiling with mvn install.

There are no errors when I run `mvn install -DskipTests` on the `endocode/compliancePlugin` (#301) branch (7 months old).

```
[INFO] Reactor Summary:
[INFO] 
[INFO] parent 0.0.1-SNAPSHOT .............................. SUCCESS [  0.513 s]
[INFO] core 0.0.1-SNAPSHOT ................................ SUCCESS [  6.268 s]
[INFO] analyzer 0.0.1-SNAPSHOT ............................ SUCCESS [  0.228 s]
[INFO] javacg-opal 0.0.1-SNAPSHOT ......................... SUCCESS [ 10.724 s]
[INFO] server 0.0.1-SNAPSHOT .............................. SUCCESS [  9.567 s]
[INFO] metadata-plugin 0.0.1-SNAPSHOT ..................... SUCCESS [  8.827 s]
[INFO] graph-plugin 0.0.1-SNAPSHOT ........................ SUCCESS [  4.172 s]
[INFO] pom-analyzer 0.0.1-SNAPSHOT ........................ SUCCESS [  8.835 s]
[INFO] repo-cloner-plugin 0.0.1-SNAPSHOT .................. SUCCESS [  3.331 s]
[INFO] quality-analyzer 0.0.1-SNAPSHOT .................... SUCCESS [  8.763 s]
[INFO] restapi-plugin 0.1.0-SNAPSHOT ...................... SUCCESS [ 10.736 s]
[INFO] vulnerability-consumer 0.0.1-SNAPSHOT .............. SUCCESS [  8.904 s]
[INFO] repo-analyzer 0.0.1-SNAPSHOT ....................... SUCCESS [  8.886 s]
[INFO] vulnerability-cache-invalidation-plugin 0.0.1-SNAPSHOT SUCCESS [  8.609 s]
[INFO] vulnerability-cache-processor-plugin 0.0.1-SNAPSHOT  SUCCESS [  4.205 s]
[INFO] license-detector 0.0.1-SNAPSHOT .................... SUCCESS [  8.808 s]
[INFO] license-feeder 0.0.1-SNAPSHOT ...................... SUCCESS [  8.689 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:00 min
[INFO] Finished at: 2022-02-23T16:48:02+01:00
[INFO] ------------------------------------------------------------------------

```

When I do the same on the new branch #415, there are errors related to the implementation of the Kafka plugin.

```
[INFO] vulnerability-packages-listener 0.0.1-SNAPSHOT ..... SUCCESS [  9.119 s]
[INFO] dummy-plugin 0.0.1-SNAPSHOT ........................ SUCCESS [  4.828 s]
[INFO] debian-license-detector 0.0.1-SNAPSHOT ............. SUCCESS [  9.351 s]
[INFO] license-detector 0.0.1-SNAPSHOT .................... FAILURE [  0.119 s]
[INFO] license-feeder 0.0.1-SNAPSHOT ...................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:06 min
[INFO] Finished at: 2022-02-23T17:10:17+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project license-detector: Compilation failure: Compilation failure: 
[ERROR] /home/michelescarlato/gitrepo/fasten/analyzer/license-detector/src/main/java/eu/fasten/analyzer/licensedetector/LicenseDetectorPlugin.java:[41,19] eu.fasten.analyzer.licensedetector.LicenseDetectorPlugin.LicenseDetector is not abstract and does not override abstract method setTopics(java.util.List<java.lang.String>) in eu.fasten.core.plugins.KafkaPlugin
[ERROR] /home/michelescarlato/gitrepo/fasten/analyzer/license-detector/src/main/java/eu/fasten/analyzer/licensedetector/LicenseDetectorPlugin.java:[62,9] method does not override or implement a method from a supertype
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :license-detector

```

I also tried to change line 41, from `public static class`  into `abstract class`, and received this error:

```
[INFO] dummy-plugin 0.0.1-SNAPSHOT ........................ SUCCESS [  4.777 s]
[INFO] debian-license-detector 0.0.1-SNAPSHOT ............. SUCCESS [  9.977 s]
[INFO] license-detector 0.0.1-SNAPSHOT .................... FAILURE [  0.143 s]
[INFO] license-feeder 0.0.1-SNAPSHOT ...................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:08 min
[INFO] Finished at: 2022-02-23T17:07:24+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project license-detector: Compilation failure
[ERROR] /home/michelescarlato/gitrepo/fasten/analyzer/license-detector/src/main/java/eu/fasten/analyzer/licensedetector/LicenseDetectorPlugin.java:[62,9] method does not override or implement a method from a supertype
[ERROR] 
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :license-detector

```

And this is the error related to the `license-feeder`:

```
[INFO] vulnerability-cache-invalidation-plugin 0.0.1-SNAPSHOT SUCCESS [  9.357 s]
[INFO] parallel-vulnerability-cache-invalidation-plugin 0.0.1-SNAPSHOT SUCCESS [  9.297 s]
[INFO] vulnerability-statements-processor 0.0.1-SNAPSHOT .. SUCCESS [  9.377 s]
[INFO] vulnerability-packages-listener 0.0.1-SNAPSHOT ..... SUCCESS [  8.952 s]
[INFO] dummy-plugin 0.0.1-SNAPSHOT ........................ SUCCESS [  4.664 s]
[INFO] debian-license-detector 0.0.1-SNAPSHOT ............. SUCCESS [  9.383 s]
[INFO] license-feeder 0.0.1-SNAPSHOT ...................... FAILURE [  0.157 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:07 min
[INFO] Finished at: 2022-02-23T17:02:04+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project license-feeder: Compilation failure: Compilation failure: 
[ERROR] /home/michelescarlato/gitrepo/fasten/analyzer/license-feeder/src/main/java/eu/fasten/analyzer/licensefeeder/LicenseFeederPlugin.java:[31,19] eu.fasten.analyzer.licensefeeder.LicenseFeederPlugin.LicenseFeeder is not abstract and does not override abstract method setTopics(java.util.List<java.lang.String>) in eu.fasten.core.plugins.KafkaPlugin
[ERROR] /home/michelescarlato/gitrepo/fasten/analyzer/license-feeder/src/main/java/eu/fasten/analyzer/licensefeeder/LicenseFeederPlugin.java:[195,9] method does not override or implement a method from a supertype
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :license-feeder
```

Did you already see this error?

**Comment by mir-am** [comment]
I think the error is easy to fix. You need to change the signature of method `setTopics` as stated in the error. Take a look at the Kafka plugin interface (`eu.fasten.core.plugins.KafkaPlugin`) and also another plug-in (e.g. `OPAL`) to see how the method `setTopics` is implemented. It should accept `List<String>` and set Kafka topics in the plugin.

**Comment by michelescarlato** [comment]
[This is the GitHub action](https://github.com/fasten-project/fasten/runs/5317265437?check_suite_focus=true):


```
[INFO] parent ............................................. SUCCESS [01:13 min]
16823
[INFO] core ............................................... SUCCESS [01:41 min]
16824
[INFO] analyzer ........................................... SUCCESS [  2.594 s]
16825
[INFO] server ............................................. SUCCESS [ 27.077 s]
16826
[INFO] license-detector ................................... FAILURE [  0.389 s]
16827
[INFO] ------------------------------------------------------------------------
16828
[INFO] BUILD FAILURE
16829
[INFO] ------------------------------------------------------------------------
16830
[INFO] Total time:  03:24 min
16831
[INFO] Finished at: 2022-02-24T10:25:31Z
16832
[INFO] ------------------------------------------------------------------------
16833
Error:  Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.2:test (default-test) on project license-detector: No tests were executed!  (Set -DfailIfNoTests=false to ignore this error.) -> [Help 1]
16834
Error:  
16835
Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.
16836
Error:  Re-run Maven using the -X switch to enable full debug logging.
16837
Error:  
16838
Error:  For more information about the errors and possible solutions, please read the following articles:
16839
Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
16840
Error:  
16841
Error:  After correcting the problems, you can resume the build with the command
16842
Error:    mvn <args> -rf :license-detector
16843
Error: Process completed with exit code 1.
```





This is locally, using :

`mvn -B -ntp clean install -rf :license-detector`


```
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running eu.fasten.analyzer.licensefeeder.LicenseFeederIntegrationTest
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [jar:file:/home/michelescarlato/.m2/repository/org/slf4j/slf4j-nop/1.7.30/slf4j-nop-1.7.30.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [jar:file:/home/michelescarlato/.m2/repository/ch/qos/logback/logback-classic/1.2.7/logback-classic-1.2.7.jar!/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [org.slf4j.helpers.NOPLoggerFactory]
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 6.706 s - in eu.fasten.analyzer.licensefeeder.LicenseFeederIntegrationTest
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] 
[INFO] --- jacoco-maven-plugin:0.8.2:report (report) @ license-feeder ---
[INFO] Loading execution data file /home/michelescarlato/gitrepo/fasten/analyzer/license-feeder/target/jacoco.exec
[INFO] Analyzed bundle 'license-feeder' with 2 classes
[INFO] 
[INFO] --- maven-jar-plugin:2.4:jar (default-jar) @ license-feeder ---
[INFO] Building jar: /home/michelescarlato/gitrepo/fasten/analyzer/license-feeder/target/license-feeder-0.0.1-SNAPSHOT.jar
[INFO] 
[INFO] --- maven-shade-plugin:3.2.1:shade (default) @ license-feeder ---
[INFO] Including eu.fasten:core:jar:0.0.1-SNAPSHOT in the shaded jar.
[INFO] Including it.unimi.dsi:dsiutils:jar:2.6.17 in the shaded jar.
[INFO] Including it.unimi.di:jsap:jar:20210129 in the shaded jar.
[INFO] Including org.apache.commons:commons-configuration2:jar:2.7 in the shaded jar.
[INFO] Including org.apache.commons:commons-text:jar:1.8 in the shaded jar.
[INFO] Including commons-logging:commons-logging:jar:1.2 in the shaded jar.
[INFO] Including org.apache.commons:commons-math3:jar:3.6.1 in the shaded jar.
[INFO] Including com.google.guava:guava:jar:30.1-jre in the shaded jar.
[INFO] Including com.google.guava:failureaccess:jar:1.0.1 in the shaded jar.
[INFO] Including com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava in the shaded jar.
[INFO] Including com.google.code.findbugs:jsr305:jar:3.0.2 in the shaded jar.
[INFO] Including com.google.errorprone:error_prone_annotations:jar:2.3.4 in the shaded jar.
[INFO] Including com.google.j2objc:j2objc-annotations:jar:1.3 in the shaded jar.
[INFO] Including it.unimi.dsi:sux4j:jar:5.2.3 in the shaded jar.
[INFO] Including it.unimi.dsi:fastutil:jar:8.5.2 in the shaded jar.
[INFO] Including it.unimi.dsi:webgraph:jar:3.6.10 in the shaded jar.
[INFO] Including net.sf.jung:jung-api:jar:2.1.1 in the shaded jar.
[INFO] Including net.sf.jung:jung-io:jar:2.1.1 in the shaded jar.
[INFO] Including net.sf.jung:jung-algorithms:jar:2.1.1 in the shaded jar.
[INFO] Including commons-io:commons-io:jar:2.8.0 in the shaded jar.
[INFO] Including org.apache.commons:commons-lang3:jar:3.11 in the shaded jar.
[INFO] Including commons-lang:commons-lang:jar:2.6 in the shaded jar.
[INFO] Including org.eclipse.jdt:org.eclipse.jdt.annotation:jar:2.0.0 in the shaded jar.
[INFO] Including it.unimi.dsi:law:jar:2.7.2 in the shaded jar.
[INFO] Including org.slf4j:log4j-over-slf4j:jar:1.7.30 in the shaded jar.
[INFO] Including org.slf4j:jcl-over-slf4j:jar:1.7.30 in the shaded jar.
[INFO] Including it.unimi.dsi:webgraph-big:jar:3.6.5 in the shaded jar.
[INFO] Including com.martiansoftware:jsap:jar:2.1 in the shaded jar.
[INFO] Including org.eclipse.jgit:org.eclipse.jgit:jar:5.2.1.201812262042-r in the shaded jar.
[INFO] Including com.jcraft:jsch:jar:0.1.54 in the shaded jar.
[INFO] Including com.jcraft:jzlib:jar:1.1.1 in the shaded jar.
[INFO] Including com.googlecode.javaewah:JavaEWAH:jar:1.1.6 in the shaded jar.
[INFO] Including org.json:json:jar:20180813 in the shaded jar.
[INFO] Including org.apache.kafka:kafka-clients:jar:2.8.1 in the shaded jar.
[INFO] Including com.github.luben:zstd-jni:jar:1.4.9-1 in the shaded jar.
[INFO] Including org.lz4:lz4-java:jar:1.7.1 in the shaded jar.
[INFO] Including org.xerial.snappy:snappy-java:jar:1.1.8.1 in the shaded jar.
[INFO] Including com.esotericsoftware:kryo:jar:5.0.0-RC4 in the shaded jar.
[INFO] Including com.esotericsoftware:reflectasm:jar:1.11.8 in the shaded jar.
[INFO] Including com.esotericsoftware:minlog:jar:1.3.1 in the shaded jar.
[INFO] Including org.postgresql:postgresql:jar:42.2.25 in the shaded jar.
[INFO] Including org.checkerframework:checker-qual:jar:3.5.0 in the shaded jar.
[INFO] Including org.rocksdb:rocksdbjni:jar:6.15.2 in the shaded jar.
[INFO] Including com.github.t9t.jooq:jooq-postgresql-json:jar:1.0.0 in the shaded jar.
[INFO] Including org.jgrapht:jgrapht-core:jar:1.5.0 in the shaded jar.
[INFO] Including org.jheaps:jheaps:jar:0.13 in the shaded jar.
[INFO] Including org.jgrapht:jgrapht-opt:jar:1.5.0 in the shaded jar.
[INFO] Skipping pom dependency org.jboss.shrinkwrap.resolver:shrinkwrap-resolver-depchain:pom:3.1.4 in the shaded jar.
[INFO] Including org.jboss.shrinkwrap.resolver:shrinkwrap-resolver-api:jar:3.1.4 in the shaded jar.
[INFO] Including org.jboss.shrinkwrap.resolver:shrinkwrap-resolver-spi:jar:3.1.4 in the shaded jar.
[INFO] Including org.jboss.shrinkwrap.resolver:shrinkwrap-resolver-api-maven:jar:3.1.4 in the shaded jar.
[INFO] Including org.jboss.shrinkwrap.resolver:shrinkwrap-resolver-spi-maven:jar:3.1.4 in the shaded jar.
[INFO] Including org.jboss.shrinkwrap.resolver:shrinkwrap-resolver-api-maven-archive:jar:3.1.4 in the shaded jar.
[INFO] Including org.jboss.shrinkwrap:shrinkwrap-api:jar:1.2.6 in the shaded jar.
[INFO] Including org.jboss.shrinkwrap.resolver:shrinkwrap-resolver-impl-maven-archive:jar:3.1.4 in the shaded jar.
[INFO] Including org.jboss.shrinkwrap:shrinkwrap-impl-base:jar:1.2.6 in the shaded jar.
[INFO] Including org.jboss.shrinkwrap:shrinkwrap-spi:jar:1.2.6 in the shaded jar.
[INFO] Including org.jboss.shrinkwrap.resolver:shrinkwrap-resolver-spi-maven-archive:jar:3.1.4 in the shaded jar.
[INFO] Including org.eclipse.sisu:org.eclipse.sisu.plexus:jar:0.3.4 in the shaded jar.
[INFO] Including org.codehaus.plexus:plexus-compiler-javac:jar:2.7 in the shaded jar.
[INFO] Including org.codehaus.plexus:plexus-compiler-api:jar:2.7 in the shaded jar.
[INFO] Including org.jboss.shrinkwrap.resolver:shrinkwrap-resolver-api-maven-embedded:jar:3.1.4 in the shaded jar.
[INFO] Including org.jboss.shrinkwrap.resolver:shrinkwrap-resolver-impl-maven-embedded:jar:3.1.4 in the shaded jar.
[INFO] Including org.arquillian.spacelift:arquillian-spacelift:jar:1.0.2 in the shaded jar.
[INFO] Including org.arquillian.spacelift:arquillian-spacelift-api:jar:1.0.2 in the shaded jar.
[INFO] Including org.apache.commons:commons-compress:jar:1.21 in the shaded jar.
[INFO] Including org.jboss.shrinkwrap.resolver:shrinkwrap-resolver-impl-maven-patched:jar:3.1.4 in the shaded jar.
[INFO] Including org.apache.maven:maven-resolver-provider:jar:3.6.3 in the shaded jar.
[INFO] Including javax.inject:javax.inject:jar:1 in the shaded jar.
[INFO] Including org.apache.maven:maven-model:jar:3.6.3 in the shaded jar.
[INFO] Including org.apache.maven:maven-model-builder:jar:3.6.3 in the shaded jar.
[INFO] Including org.apache.maven:maven-builder-support:jar:3.6.3 in the shaded jar.
[INFO] Including org.eclipse.sisu:org.eclipse.sisu.inject:jar:0.3.4 in the shaded jar.
[INFO] Including org.apache.maven:maven-repository-metadata:jar:3.6.3 in the shaded jar.
[INFO] Including org.apache.maven:maven-settings:jar:3.6.3 in the shaded jar.
[INFO] Including org.apache.maven:maven-settings-builder:jar:3.6.3 in the shaded jar.
[INFO] Including org.apache.maven.resolver:maven-resolver-api:jar:1.4.1 in the shaded jar.
[INFO] Including org.apache.maven.resolver:maven-resolver-impl:jar:1.4.1 in the shaded jar.
[INFO] Including org.apache.maven.resolver:maven-resolver-spi:jar:1.4.1 in the shaded jar.
[INFO] Including org.apache.maven.resolver:maven-resolver-util:jar:1.4.1 in the shaded jar.
[INFO] Including org.apache.maven.resolver:maven-resolver-connector-basic:jar:1.4.1 in the shaded jar.
[INFO] Including org.apache.maven.resolver:maven-resolver-transport-wagon:jar:1.4.1 in the shaded jar.
[INFO] Including org.codehaus.plexus:plexus-interpolation:jar:1.25 in the shaded jar.
[INFO] Including org.codehaus.plexus:plexus-utils:jar:3.2.1 in the shaded jar.
[INFO] Including org.sonatype.plexus:plexus-sec-dispatcher:jar:1.4 in the shaded jar.
[INFO] Including org.sonatype.plexus:plexus-cipher:jar:1.4 in the shaded jar.
[INFO] Including org.apache.maven.wagon:wagon-provider-api:jar:3.3.4 in the shaded jar.
[INFO] Including org.apache.maven.wagon:wagon-file:jar:3.3.4 in the shaded jar.
[INFO] Including org.apache.maven.wagon:wagon-http-lightweight:jar:2.12 in the shaded jar.
[INFO] Including org.apache.maven.wagon:wagon-http-shared:jar:2.12 in the shaded jar.
[INFO] Including org.jsoup:jsoup:jar:1.14.2 in the shaded jar.
[INFO] Including org.apache.maven:maven-artifact:jar:3.6.3 in the shaded jar.
[INFO] Including commons-codec:commons-codec:jar:1.15 in the shaded jar.
[INFO] Including com.squareup.okhttp3:okhttp:jar:4.9.2 in the shaded jar.
[INFO] Including com.squareup.okio:okio:jar:2.8.0 in the shaded jar.
[INFO] Including org.jetbrains.kotlin:kotlin-stdlib-common:jar:1.4.0 in the shaded jar.
[INFO] Including org.jetbrains.kotlin:kotlin-stdlib:jar:1.4.10 in the shaded jar.
[INFO] Including org.jetbrains:annotations:jar:13.0 in the shaded jar.
[INFO] Including org.apache.maven.shared:maven-invoker:jar:3.0.1 in the shaded jar.
[INFO] Including org.apache.maven.shared:maven-shared-utils:jar:3.2.1 in the shaded jar.
[INFO] Including org.codehaus.plexus:plexus-component-annotations:jar:1.7.1 in the shaded jar.
[INFO] Including com.google.code.gson:gson:jar:2.8.6 in the shaded jar.
[INFO] Including org.pf4j:pf4j:jar:3.4.0 in the shaded jar.
[INFO] Including com.github.zafarkhaja:java-semver:jar:0.9.0 in the shaded jar.
[INFO] Including info.picocli:picocli:jar:4.0.4 in the shaded jar.
[INFO] Excluding org.slf4j:slf4j-api:jar:1.7.32 from the shaded jar.
[INFO] Including ch.qos.logback:logback-classic:jar:1.2.7 in the shaded jar.
[INFO] Including ch.qos.logback:logback-core:jar:1.2.7 in the shaded jar.
[INFO] Including org.objenesis:objenesis:jar:3.2 in the shaded jar.
[WARNING] jsap-2.1.jar, jsap-20210129.jar define 69 overlapping classes: 
[WARNING]   - com.martiansoftware.jsap.StringParser
[WARNING]   - com.martiansoftware.jsap.SyntaxException
[WARNING]   - com.martiansoftware.jsap.JSAPResult
[WARNING]   - com.martiansoftware.jsap.Parameter
[WARNING]   - com.martiansoftware.jsap.Option
[WARNING]   - com.martiansoftware.jsap.xml.JSAPConfig
[WARNING]   - com.martiansoftware.jsap.IllegalMultipleDeclarationException
[WARNING]   - com.martiansoftware.jsap.stringparsers.ForNameStringParser
[WARNING]   - com.martiansoftware.jsap.stringparsers.IntSizeStringParser
[WARNING]   - com.martiansoftware.jsap.ant.ParameterConfiguration
[WARNING]   - 59 more...
[WARNING] picocli-4.0.4.jar, kotlin-stdlib-1.4.10.jar define 1 overlapping classes: 
[WARNING]   - META-INF.versions.9.module-info
[WARNING] jcl-over-slf4j-1.7.30.jar, commons-logging-1.2.jar define 6 overlapping classes: 
[WARNING]   - org.apache.commons.logging.impl.SimpleLog$1
[WARNING]   - org.apache.commons.logging.Log
[WARNING]   - org.apache.commons.logging.impl.SimpleLog
[WARNING]   - org.apache.commons.logging.LogConfigurationException
[WARNING]   - org.apache.commons.logging.impl.NoOpLog
[WARNING]   - org.apache.commons.logging.LogFactory
[WARNING] maven-shade-plugin has detected that some class files are
[WARNING] present in two or more JARs. When this happens, only one
[WARNING] single version of the class is copied to the uber jar.
[WARNING] Usually this is not harmful and you can skip these warnings,
[WARNING] otherwise try to manually exclude artifacts based on
[WARNING] mvn dependency:tree -Ddetail=true and the above output.
[WARNING] See http://maven.apache.org/plugins/maven-shade-plugin/
[INFO] Attaching shaded artifact.
[INFO] 
[INFO] --- maven-install-plugin:2.4:install (default-install) @ license-feeder ---
[INFO] Installing /home/michelescarlato/gitrepo/fasten/analyzer/license-feeder/target/license-feeder-0.0.1-SNAPSHOT.jar to /home/michelescarlato/.m2/repository/eu/fasten/analyzer/license-feeder/0.0.1-SNAPSHOT/license-feeder-0.0.1-SNAPSHOT.jar
[INFO] Installing /home/michelescarlato/gitrepo/fasten/analyzer/license-feeder/pom.xml to /home/michelescarlato/.m2/repository/eu/fasten/analyzer/license-feeder/0.0.1-SNAPSHOT/license-feeder-0.0.1-SNAPSHOT.pom
[INFO] Installing /home/michelescarlato/gitrepo/fasten/docker/plugins/license-feeder-0.0.1-SNAPSHOT-with-dependencies.jar to /home/michelescarlato/.m2/repository/eu/fasten/analyzer/license-feeder/0.0.1-SNAPSHOT/license-feeder-0.0.1-SNAPSHOT-with-dependencies.jar
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for license-detector 0.0.1-SNAPSHOT:
[INFO] 
[INFO] license-detector ................................... SUCCESS [ 12.610 s]
[INFO] license-feeder ..................................... SUCCESS [ 18.413 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  31.210 s
[INFO] Finished at: 2022-02-24T11:33:58+01:00
[INFO] ------------------------------------------------------------------------

```

And, as I think you already know, locally, if I run the command without   `-rf :license-detector`, namely with:
`mvn -B -ntp clean install`
it breaks with the `restapi-rapid` plugin.

```
[INFO] repo-cloner-plugin 0.0.1-SNAPSHOT .................. SUCCESS [ 11.790 s]
[INFO] quality-analyzer 0.0.1-SNAPSHOT .................... SUCCESS [ 10.853 s]
[INFO] restapi-plugin 0.1.0-SNAPSHOT ...................... FAILURE [  6.891 s]
[INFO] repo-analyzer 0.0.1-SNAPSHOT ....................... SKIPPED
[INFO] vulnerability-cache-invalidation-plugin 0.0.1-SNAPSHOT SKIPPED
[INFO] parallel-vulnerability-cache-invalidation-plugin 0.0.1-SNAPSHOT SKIPPED
[INFO] vulnerability-statements-processor 0.0.1-SNAPSHOT .. SKIPPED
[INFO] vulnerability-packages-listener 0.0.1-SNAPSHOT ..... SKIPPED
[INFO] dummy-plugin 0.0.1-SNAPSHOT ........................ SKIPPED
[INFO] debian-license-detector 0.0.1-SNAPSHOT ............. SKIPPED
[INFO] license-detector 0.0.1-SNAPSHOT .................... SKIPPED
[INFO] license-feeder 0.0.1-SNAPSHOT ...................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:02 min
[INFO] Finished at: 2022-02-24T12:02:41+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.2:test (default-test) on project restapi-plugin: There are test failures.
[ERROR] 
[ERROR] Please refer to /home/michelescarlato/gitrepo/fasten/analyzer/restapi-plugin/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]

```


Important to notice that the [license-feeder is not breaking](https://github.com/fasten-project/fasten/runs/5317265446?check_suite_focus=true).

**Comment by michelescarlato** [comment]
@mir-am 

I changed in both yamls, feeder and detector, from:

`mvn clean install --projects server,:license-detector --also-make -Dtest=\!MergerEfficiencyTests#localMergerEfficiencyTest`

and:

`mvn clean install --projects server,:license-feeder --also-make -Dtest=\!MergerEfficiencyTests#localMergerEfficiencyTest`

to:

mvn -B -ntp clean install

To have the build completed I had to comment the reastapi module [here](https://github.com/fasten-project/fasten/blob/license-detector-and-feeder/analyzer/pom.xml#L43).

Now the GitHub actions seem good.

One build is now failing: [windows 2019](https://github.com/fasten-project/fasten/runs/5320025340?check_suite_focus=true) one.

I have still to test both plugins, and AFAIK to build the complete image of the license-detector I should still add a GitHub action that compile [this dockerfile](https://github.com/fasten-project/fasten/blob/license-detector-and-feeder/docker/license-detector/Dockerfile).

**Comment by michelescarlato** [comment]
@mir-am 
> One build is now failing: [windows 2019](https://github.com/fasten-project/fasten/runs/5320025340?check_suite_focus=true) one.
... what about this?

**Comment by michelescarlato** [comment]
@mir-am if you are ok with that building failing, we would like to merge this PR https://github.com/fasten-project/fasten/pull/415

**Comment by mir-am** [comment]
> @mir-am
> 
> > One build is now failing: [windows 2019](https://github.com/fasten-project/fasten/runs/5320025340?check_suite_focus=true) one.
> 
> ... what about this?

Maybe, @MagielBruntink knows why the Windows build fails here, though the Ubuntu build passes.

**Comment by michelescarlato** [comment]
fixed by @MagielBruntink in #415
